theano/compile/builders.py

```diff
@@ -137,6 +137,46 @@ def perform(self, node, inputs, outputs):
             # we wont need this copy anymore
             output[0] = variable.copy()
 
+    def c_code(self, node, name, inputs, outputs, sub):
+        order = io_toposort(self.new_inputs, self.new_outputs)
```

Comment:
This will make you use the not optimized function. use self.fn.inputs and self.fn.outputs.